### DISCUSSION TITLE
Multi-line HTML comments in partial override ignored

### DISCUSSION THREAD

**Discussion by do-me** [mkdocs/mkdocs#3183] [Q&A]
It seems like the recent Jinja2 update is causing issues with multi-line comments. 

tl;dr:

The announcement bar below is still displayed.

```html
<!-- Elements added to main will be displayed on all pages -->
{% extends "base.html" %}

<!-- Announcement bar if needed
{% block announce %}
  <style>.md-announce a,.md-announce a:focus,.md-announce a:hover{color:currentColor}.md-announce strong{white-space:nowrap}.md-announce .twitter{margin-left:.2em;color:#00acee}</style>
    <strong>Test</strong>
{% endblock %} -->
```

Please see [this issue](https://github.com/squidfunk/mkdocs-material/issues/5294) I created in the mkdocs-material repo for more info and reproduction steps.

**Comment by pawamoy** [comment]
This is expected IMO: Jinja templating takes precedence over HTML. You cannot comment Jinja blocks with HTML comments. For this you have to use Jinja comments: `{# this is commented and won't be rendered #}`.

**Comment by nejch** [comment]
I agree, if you use an HTML comment it's expected to show in the HTML source and could be using jinja logic.

It also feels dirty to use comments for this, you could use a configuration option or even environment variables to toggle the announcement bar (I assume this is mkdocs-material stuff).

```yaml
extra:
  announcement: true # or !ENV [SHOW_ANNOUNCEMENT, false]
```

```jinja2
<!-- Elements added to main will be displayed on all pages -->
{% extends "base.html" %}

{% block announce %}
  {% if config.extra.announcement %}
    <style>.md-announce a,.md-announce a:focus,.md-announce a:hover{color:currentColor}.md-announce strong{white-space:nowrap}.md-announce .twitter{margin-left:.2em;color:#00acee}</style>
      <strong>Test</strong>
  {% endif %}
{% endblock %} 
```

Or even put the the announcement text itself into config/environment variable.
